### PR TITLE
Post Show Page and Social Media Buttons

### PR DESCRIPTION
**Pull request by mattLummus**
Social Media Buttons included on application.html.haml and posts.index.html.haml.
We weren't sure about how we were including the custom url for the share buttons on the index page.

### CODE REVIEW

File: app/views/posts/index.html.haml [hunk 1]
```diff
@@ -6,5 +6,14 @@
       squmbd:
       = post.content
       = post.created_at
+      <div class="addthis_toolbox addthis_default_style " addthis:url="#{request.domain}/posts/#{post.id}">
```
Comment (elizamarcum) [inline app/views/posts/index.html.haml]: Please pull this out into a partial (e.g. `app/views/posts/_share.html.haml`) and use this on both posts index and posts show.

Also, I prefer that the div be converted to haml, since it is is unintuitive to read right now.  See http://haml.info/docs/yardoc/file.REFERENCE.html#html5_custom_data_attributes

File: spec/features/user_profile_page_spec.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 require 'spec_helper'
 
 feature "User visits profile page" do
-  let!(:bob) {Fabricate(:user, email: "bob@example.com", username: "bob")}
-  let!(:matt) {Fabricate(:user, email: "mknicos@gmail.com", username: "matt")}
+  let!(:bob) {Fabricate(:user, email: "bob@example.com", username: "bob", password: "password")}
```
Comment (elizamarcum) [inline spec/features/user_profile_page_spec.rb:4]: Why did you change the fabricate to include "password"?

File: app/models/user.rb [hunk 3]
```diff
@@ -8,12 +8,16 @@ class User < ActiveRecord::Base
          :omniauthable, :omniauth_providers => [:github]
 
   has_many :posts
-
-  validates :username, uniqueness: true
+  validates :username, uniqueness: true,
```
Comment (elizamarcum) [inline app/models/user.rb:11]: There should be a corresponding model test for this.  Take a look at `should validates_format_of` from shoulda.

File: spec/features/user_profile_page_spec.rb (hunk 2 above)
Comment (mattLummus) [inline spec/features/user_profile_page_spec.rb:4]: We were previously having an issue with users being created by Fabricate and wanted to make sure a lack of password wasn't causing it.

File: app/models/user.rb (hunk 3 above)
Comment (mattLummus) [inline app/models/user.rb:11]: We updated the user_spec to test for this.

File: Gemfile [hunk 4]
```diff
@@ -37,4 +37,5 @@ group :test do
   gem 'shoulda'
   gem 'vcr'
   gem 'webmock'
+  gem 'shoulda-matchers'
```
Comment (elizamarcum) [inline Gemfile]: The shoulda gem depends on shoulda-matchers, so you shouldn't need to add shoulda-matchers independently.

File: spec/features/user_profile_page_spec.rb (hunk 2 above)
Comment (elizamarcum) [inline spec/features/user_profile_page_spec.rb:4]: The user fabricator (spec/fabricators/user_fabricator.rb) does include password, so this change shouldn't be necessary.

I suspect that you were running into issues with the test database not getting cleaned up by rspec.

File: Gemfile (hunk 4 above)
Comment (mattLummus) [inline Gemfile]: Removed it from Gemfile.

File: app/views/posts/index.html.haml (hunk 1 above)
Comment (mattLummus) [inline app/views/posts/index.html.haml]: Moved share buttons into partials (views/social_buttons). One is for the posts index and one is for all other pages in the layout.

File: app/views/social_buttons/_index_buttons.html.haml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+.addthis_toolbox.addthis_default_style{ "addthis:url" => "#{request.domain}/posts/#{post_id}" }
```
Comment (elizamarcum) [inline app/views/social_buttons/_index_buttons.html.haml]: I believe that this partial and the _layou_buttons partial can be merged.

I also think that this should be "app/views/shared/_social_buttons.html.haml"

And the url should be generated via the rails url helpers: `posts_url(post)`, rather than manually generated.
